client/client.go

```diff
@@ -574,6 +574,56 @@ func (r *NotaryRepository) GetDelegationRoles() ([]*data.Role, error) {
 	return allDelegations, nil
 }
 
+// RoleWithSignatures is a Role with its associated signatures
+type RoleWithSignatures struct {
+	Signatures []data.Signature
+	data.Role
+}
+
+// GetRepoRoleMetaInfo returns a list of RoleWithSignatures objects for this repo
+// This represents the latest metadata for each role in this repo
+func (r *NotaryRepository) GetRepoRoleMetaInfo() ([]RoleWithSignatures, error) {
```

Comment:
@endophage If we want to display the combined results though, we should probably provide a singular function that calls `Update` once, since if we call the 2 separate functions, the results might be inconsistent (the state of the repo may have changed in between ListTargets and ListRoles on the server).  Or otherwise call one, take the repo offline, and then call the other.